internal/storage/const.go

```diff
@@ -1,44 +1,37 @@
 package storage
 
-import "fmt"
+import (
+	"fmt"
+)
+
+const storageSchemaCurrentVersion = 2
+const storageSchemaUpgradeMessage = "Storage schema upgraded to v"
+const storageSchemaUpgradeErrorText = "storage schema upgrade failed at v"
 
 // Keep table names in lower case because some DB does not support upper case.
-const preferencesTableName = "user_preferences"
+const userPreferencesTableName = "user_preferences"
 const identityVerificationTokensTableName = "identity_verification_tokens"
 const totpSecretsTableName = "totp_secrets"
 const u2fDeviceHandlesTableName = "u2f_devices"
 const authenticationLogsTableName = "authentication_logs"
+const configTableName = "config"
 
-// SQLCreateUserPreferencesTable common SQL query to create user_preferences table.
-var SQLCreateUserPreferencesTable = fmt.Sprintf(`
-CREATE TABLE IF NOT EXISTS %s (
-	username VARCHAR(100) PRIMARY KEY,
-	second_factor_method VARCHAR(11)
-)`, preferencesTableName)
-
-// SQLCreateIdentityVerificationTokensTable common SQL query to create identity_verification_tokens table.
-var SQLCreateIdentityVerificationTokensTable = fmt.Sprintf(`
-CREATE TABLE IF NOT EXISTS %s (token VARCHAR(512))
-`, identityVerificationTokensTableName)
-
-// SQLCreateTOTPSecretsTable common SQL query to create totp_secrets table.
-var SQLCreateTOTPSecretsTable = fmt.Sprintf(`
-CREATE TABLE IF NOT EXISTS %s (username VARCHAR(100) PRIMARY KEY, secret VARCHAR(64))
-`, totpSecretsTableName)
-
-// SQLCreateU2FDeviceHandlesTable common SQL query to create u2f_device_handles table.
-var SQLCreateU2FDeviceHandlesTable = fmt.Sprintf(`
-CREATE TABLE IF NOT EXISTS %s (
-	username VARCHAR(100) PRIMARY KEY,
-	keyHandle TEXT,
-	publicKey TEXT
-)`, u2fDeviceHandlesTableName)
+// sqlUpgradeCreateTableStatements is a map of the schema version number, plus a map of the table name and the statement used to create it.
+// The statement is fmt.Sprintf'd with the table name as the first argument.
+var sqlUpgradeCreateTableStatements = map[int]map[string]string{
```

Comment:
Can we create a dedicated type for version?  It is basically an int but at the call site it would be clearer. I was unsure when I saw `sqlUpgradeCreateTableStatements[1]` until I re-read this type definition. With my suggestion that statement would become `sqlUpgradeCreateTableStatements[Version(1)]`

Reply:
Done locally. Also added a ToString() method for it.